packages/inter-protocol/src/clientSupport.js

```diff
@@ -204,28 +204,67 @@ const makePsmSwapOffer = (instance, brands, opts) => {
 };
 
 /**
- * @param {Record<string, Brand>} _brands
+ * @param {{
+ *   brand: Record<string, Brand>,
+ *   vbankAsset: Record<string, { brand: Brand, displayInfo: DisplayInfo }>,
+ * }} agoricNames
+ * @param {(msg: string) => Error} makeError error constructor
+ * @returns {(a: string) => Amount<'nat'>}
+ */
+export const makeParseAmount =
+  (agoricNames, makeError = msg => RangeError(msg)) =>
+  opt => {
+    assert.typeof(opt, 'string', 'parseAmount expected string');
+    const m = opt.match(/^(?<value>[\d_]+(\.[\d_]+)?)(?<brand>[A-Z]\w*?)$/);
+    if (!m || !m.groups) {
+      throw makeError(`invalid amount: ${opt}`);
+    }
+    const anyBrand = agoricNames.brand[m.groups.brand];
+    if (!anyBrand) {
+      throw makeError(`unknown brand: ${m.groups.brand}`);
+    }
+    const assetDesc = Object.values(agoricNames.vbankAsset).find(
+      d => d.brand === anyBrand,
+    );
+    if (!assetDesc) {
+      throw makeError(`unknown brand: ${m.groups.brand}`);
+    }
+    const { displayInfo } = assetDesc;
+    if (!displayInfo.decimalPlaces || displayInfo.assetKind !== 'nat') {
+      throw makeError(`bad brand: ${displayInfo}`);
+    }
+    const value = BigInt(
+      Number(m.groups.value.replace(/_/g, '')) *
+        10 ** displayInfo.decimalPlaces,
+    );
+    /** @type {Brand<'nat'>} */
+    // @ts-expect-error dynamic cast
+    const natBrand = anyBrand;
+    const amt = { value, brand: natBrand };
+    return amt;
+  };
+
+/**
+ * @param {Pick<import('@agoric/vats/tools/board-utils.js').AgoricNamesRemotes, 'brand' | 'vbankAsset'>} agoricNames
  * @param {{
  *   offerId: string,
  *   give: string,
  *   maxBuy: string,
  *   wantMinimum?: string,
- *   parseAmount: (x: string) => Amount<'nat'>,
  * } & ({
  *   price: number,
  * } | {
  *   discount: number,  // -1 to 1. e.g. 0.10 for 10% discount, -0.05 for 5% markup
  * })} opts
  * @returns {import('@agoric/smart-wallet/src/offers.js').OfferSpec}
  */
-const makeBidOffer = (_brands, opts) => {
-  assert.typeof(opts.parseAmount, 'function');
+const makeBidOffer = (agoricNames, opts) => {
```

Comment:
changing the type of the 1st arg of `makeBidOffer` is a breaking API change, right?

Reply:
Arguably. It breaks the API of this function in this module. This package does not declare what its exports are. My stance is until the package declares exports, none are supported as an API for the package. If decide that anything in `src` is part of a declared API then I think we'll need some way to opt out of that because that would greatly impede refactoring.